src/main/java/mate/academy/hibernate/relations/util/HibernateUtil.java

```diff
@@ -1,9 +1,24 @@
 package mate.academy.hibernate.relations.util;
 
 import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
 
 public class HibernateUtil {
+    private static SessionFactory sessionFactory;
+
+    private HibernateUtil() {
+
+    }
+
     public static SessionFactory getSessionFactory() {
-        return null;
+        if (sessionFactory == null) {
+            try {
+                sessionFactory = new Configuration().configure().buildSessionFactory();
+            } catch (Exception e) {
+                throw new ExceptionInInitializerError("Initial SessionFactory creation failed: "
+                        + e);
+            }
```

Comment:
Consider catching the more specific Hibernate exception (e.g., `HibernateException`) instead of the generic `Exception`. This would allow for more precise exception handling and potentially provide more informative error messages to the user.